src/scenes/bon_odori.inc.c

```diff
@@ -1,81 +1,356 @@
+}
+
+void func_08020e90() {
+    func_0800186c(&gBonOdoriInfo.unk60, &D_03004b10.unk54, 0xe0, 0x10, 0x200);
+    func_0800186c(&gBonOdoriInfo.unk460, &D_03004b10.unk254, 0xe0, 0x10, 0x200);
+}
+
+void func_08020ed4() {
+    gBonOdoriInfo.unk860 = 0;
+}
+
+void func_08020ee8(void) {
+    u8 temp0;
+    u32 temp1 = gBonOdoriInfo.unk860;
+    struct struct_030055d0 *temp2;
+    
+    if (gBonOdoriInfo.unk860 != 0) {
+        gBonOdoriInfo.unk54 = 3;
+        gBonOdoriInfo.unk55 = 4;
+        func_0800bc40();
+    } else {
+        gBonOdoriInfo.unk54 = 3;
+        
+        temp2 = D_030055d0;
+        if ((u16)func_08001980(2) != 0)
+            temp0 = 5;
+        else
+            temp0 = 6;
+        temp2->gameInfo.bonOdori.unk55 = temp0;
+        
+        gBonOdoriInfo.unk40 = temp1;
+        gBonOdoriInfo.unk42 = 0;
+    }
+}
+
+void func_08020f48() {
+    if (gBonOdoriInfo.unk864 != 0) {
+        gBonOdoriInfo.unk864 -= 1;
+        if (gBonOdoriInfo.unk864 == 0) {
+            func_08017338(1, 0);
+        }
+    }
+    func_08020834();
+    if (gBonOdoriInfo.unk40 != 0)
+        gBonOdoriInfo.unk40 -= 1;
+}
+
+void func_08020f8c() {
+    func_0800e128();
+}
+
+// Cue call handler
+void func_08020f98(u32 unused_arg0, struct struct_080179f4_sub* arg1, u32 arg2) {
+    arg1->unk0.u8[0] = arg2;
+    gBonOdoriInfo.unk862 = arg2;
+}
+
+// Cue update
+u8 func_08020fb0(u32 unused_arg0, struct struct_080179f4_sub* unused_arg1, u32 arg2) {
+    u32 temp = func_0800c3a4(0x30);
+    if (arg2 > temp) {
+        return 1;
+    } else {
+        return 0;
+    }
+}
+
+// Cue end
+void func_08020fcc(u32 unused_arg0, struct struct_080179f4_sub* unused_arg1) {
+
+}
+
+// Successful cue hit
+void func_08020fd0(u32 unused_arg0, struct struct_080179f4_sub* arg1) {
+    func_080207ec(arg1->unk0.u8[0]);
+    func_08002634(&s_HC_seqData);
+}
+
+// "Barely/near" hit
+void func_08020fe8(u32 unused_arg0, struct struct_080179f4_sub* arg1) {
+    func_080207ec(arg1->unk0.u8[0]);
+    func_08002634(&s_tebyoushi_pati_seqData);
+    gBonOdoriInfo.unk42 = 1;
+}
+
+// Cue miss
+void func_0802100c(u32 unused_arg0, struct struct_080179f4_sub* unused_arg1) {
+    gBonOdoriInfo.unk860 += 1;
+    func_0800bc40();
+    gBonOdoriInfo.unk42 = 1;
+}
+
+void func_08021034() {
+    gBonOdoriInfo.unk864 = func_0800c3a4(6);
+    func_08017338(0, 0);
+    func_080207ec(gBonOdoriInfo.unk862);
+    gBonOdoriInfo.unk860 += 1;
+    func_08002634(&s_HC_seqData);
+}
+
+void func_08021084() {
+    u32 i;
+    u32* temp;
+    for (i = 0; i < 4; i++) {
+        if (gBonOdoriInfo.unk4C[i] != 0) {
+            continue;
+        }
+        temp = func_0802075c(0, i);
+        if (i <= 2 &&gBonOdoriInfo.unk54 != 0)  {
+            temp = func_0802075c(gBonOdoriInfo.unk55, i);
+        }
+        func_0804d8f8(D_03005380, gBonOdoriInfo.unk44[i], temp, 0, 1, 0x7f, 0);
+    }
+    if (gBonOdoriInfo.unk40 == 0) {
+        if (gBonOdoriInfo.unk42 != 0) {
```

Comment:
seems like unk42 is a boolean, in which case you could replace this with just (gBonOdoriInfo.unk42) and then replace all sets with TRUE or FALSE instead of 1 or 0